src/tagmanager/tm_parser.c

```diff
@@ -818,6 +818,28 @@ void tm_parser_verify_type_mappings(void)
 }
 
 
+/* determine anonymous tags from tag names only when corresponding
+ * ctags information is not available */
+gboolean tm_parser_is_anon_name(TMParserType lang, gchar *name)
+{
+	guint i;
+	char dummy;
+
+	if (lang == TM_PARSER_C || lang == TM_PARSER_CPP)
+	{
+		return sscanf(name, "anon_%*[a-z]_%u%c", &i, &dummy) == 1 ||
+			sscanf(name, "__anon%u%c", &i, &dummy) == 1;
+	}
+	else if (lang == TM_PARSER_FORTRAN || lang == TM_PARSER_F77)
+	{
+		return sscanf(name, "Structure#%u%c", &i, &dummy) == 1 ||
+			sscanf(name, "Interface#%u%c", &i, &dummy) == 1 ||
+			sscanf(name, "Enum#%u%c", &i, &dummy) == 1;
+	}
+	return FALSE;
```

Comment:
The unit test files for java script indicate that the naming is changed. Why is that not handled here (I know it also wasn't in the old `tm_tag_is_anon()` but I would like to understand why)?

Reply:
Basically we should now always get the anon flag in our tags file (notice the `1` at the end) so it should be needed only for legacy tags files. But it's true we should probably check for the `__anon` name in case we read tags in the ctags file format and this file is generated without the `E` field.

Reply:
But what about existing tag files for java script?
```diff
-AnonymousFunction2�16�(n)�class3.c3m1�0
+anon_function_1�16�()�class2.c2m1�1
```